Userland/Libraries/LibDSP/Track.cpp

```diff
@@ -64,6 +64,10 @@ bool NoteTrack::check_processor_chain_valid() const
 ErrorOr<void> Track::resize_internal_buffers_to(size_t buffer_size)
 {
     m_secondary_sample_buffer = TRY(FixedArray<Sample>::create(buffer_size));
+    bool false_variable = false;
+    while (!m_sample_lock.compare_exchange_strong(false_variable, true)) { }
+    FixedArray<Sample> cache = TRY(FixedArray<Sample>::create(buffer_size));
+    m_cached_sample_buffer.swap(cache);
     return {};
```

Comment:
Before the return you have to store "false" back into the lock atomic, otherwise using this function once will disable the sample display forever :^)

Reply:
For a bit I was confused why you didn't comment on all three occurrences until I noticed that I already restored the lock in one of them.